### DISCUSSION TITLE
How to detect if a SpriteBundle is clicked/touched

### DISCUSSION THREAD

**Discussion by awhillas** [bevyengine/bevy#8794] [Q&A]
So i have a simple game i which things appear on screen and move around and need to be touched or clicked to kill them. I can't seen anything on interaction much in the ref or guide? There is the ButtonBundle which has built in interactions, should i use this for all sprites if i want interactions?

**Comment by awhillas** [comment]
I found [this](https://stackoverflow.com/a/67009510/196732) and updated it for Bevy 0.10
```rust
fn handle_mouse_clicks(
    mouse_input: Res<Input<MouseButton>>,
    window_query: Query<&Window, With<PrimaryWindow>>,
) {
    let win = window_query.get_single().unwrap();
    if mouse_input.just_pressed(MouseButton::Left) {
        println!("click at {:?}", win.cursor_position());
    }
}
```
But this is not going to scale well i.e. I'd need to have one of these per thing that need interaction... i guess OOP has the upper hand with this stuff...?

**Comment by mockersf** [comment]
You'll need to know all the things that can be interacted with. If you only have "a few" sprites, you can use a query on the sprite and the transform, to get its size and position. Then you can do that comparison in a loop over all those sprites.

If you have more than "a few", the recommended way is to keep a quad tree or another spatial data structure, that will let you reduce the amount of comparison you will have to do.

Depending on your constraints and what other things you're doing, "a few" could easily be up to a few thousands. It also depends if you're memory or CPU constrained. There is an example `many_buttons` in Bevy repo that does "naive" mouse position comparison over 12100 buttons and runs at 240 fps on my laptop.

**Comment by Dimchikkk** [comment]
I've built simple interactive system for sprites: https://github.com/StaffEngineer/velo/blob/b72e12bc079caa68a0bbc11e94cab21bb0b32f4f/src/ui_plugin/systems/interactive_sprites.rs#L16 you can do something similar.

**Comment by NiklasEi** [comment]
If you don't want to build your own, you can use bevy_mod_picking with the sprite backend.
See https://github.com/aevyrie/bevy_mod_picking/blob/main/examples/sprite.rs
